src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,107 @@
 import './PostDetails.scss';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+export const PostDetails = ({ id, body }) => {
+  const [comments, setComments] = useState([]);
+  const [showComments, setShowComments] = useState(true);
+
+  useEffect(() => {
+    getPostComments(id)
+      .then(setComments);
+  }, [id]);
+
+  const handleShowComments = () => {
+    setShowComments(!showComments);
+  };
+
+  const handleCommentAdd = (postId, userName, userEmail, content) => {
+    addComment(postId, userName, userEmail, content)
+      .then(() => {
+        getPostComments(id)
+          .then(setComments);
+      });
+  };
+
+  const handleCommentDelete = (commentId) => {
+    deleteComment(commentId)
+      .then(() => {
+        getPostComments(id)
+          .then(setComments);
+      });
+  };
+
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+
+      <section className="PostDetails__post">
+        <p>{body}</p>
+      </section>
+
+      {comments.length > 0 && (
+        <section className="PostDetails__comments">
+          {showComments && (
+            <button
+              type="button"
+              className="button"
+              onClick={handleShowComments}
+            >
+              {
+                `Show
+                ${comments.length}
+                comments`
+              }
+            </button>
+          )}
+
+          {!showComments && (
+            <button
+              type="button"
+              className="button"
+              onClick={handleShowComments}
+            >
+              {
+                `Hide
+                ${comments.length}
+                comments`
+              }
+            </button>
+          )}
+          {!showComments && (
+            <ul className="PostDetails__list">
+              {comments.map(comment => (
+                <li className="PostDetails__list-item" key={comment.body}>
```

Comment:
```suggestion
                <li className="PostDetails__list-item" key={comment.id}>
```
comment body could be repeated